### PR TITLE
Improved error messaging for conflicting link defs in schema.ts

### PR DESCRIPTION
**Pull request by markyfyi**
The schema planner doesn't account for a sneaky invariant: what if the user has a link def, but in the _opposite direction_ of one in the database?  As in, the forward and reverse identities are swapped.  Right now, we catch this, but throw a misleading "check your schema file for dupes" error.  In this change, I add some code to check for this case, and throw a (hopefully) more informative error.

To repro - go to `sandbox/cli-nodejs/instant.schema.ts` and swap the `forward` and `reverse` props of one of the links.

### CODE REVIEW

**Review by nezaj — APPROVED** [review]
LGTM w/ one nit.

I like the informative error, Stripe goes pretty with some of their error messages and I think it's a really nice user experience

File: server/src/instant/model/schema.clj [hunk 1]
```diff
@@ -8,6 +8,8 @@
             [instant.util.exception :as ex])
   (:import (java.util UUID)))
 
+
```
Comment (nezaj) [inline server/src/instant/model/schema.clj]: Nit. extra white-space

File: server/src/instant/model/schema.clj [hunk 2]
```diff
@@ -149,21 +151,39 @@
...
-                               (when (= op :add-attr)
-                                 (attr-ident-names data)))))
-        dups (->> (concat current-ident-names ident-names)
+  (let [current-fwd-ident-names (->> current-attrs
+                                     (map attr-model/fwd-ident-name)
+                                     (map vec))
+        current-all-ident-names (->> current-attrs
+                                     (mapcat attr-ident-names)
+                                     (map vec))
+        steps-ident-names (->>
```
Comment (nezaj) [inline server/src/instant/model/schema.clj]: Nit. Might be nice to call this `new-ident-names`

**Review by stopachka — APPROVED** [review]
LGTM!

**Pull request merged**